packages/puppeteer-core/src/bidi/HTTPRequest.ts

```diff
@@ -230,12 +231,16 @@ export class BidiHTTPRequest extends HTTPRequest {
     _priority?: number
   ): Promise<void> {
     this.interception.handled = true;
-    const responseBody: string | undefined =
-      response.body && response.body instanceof Uint8Array
-        ? response.body.toString('base64')
-        : response.body
-          ? btoa(response.body)
-          : undefined;
+
+    let responseContentLength: number | undefined;
+    let responseBodyBase64: string | undefined;
+    if (response.body && isString(response.body)) {
```

Comment:
nit: let's move the if isString else isBuffer to if (response.body) so that we do not need to check the body twice (and imo it is more readable).